thirdparty/misc/patches/qoa-min-fix.patch

```diff
@@ -24,20 +24,28 @@ index 592082933a..c890b88bd6 100644
  
  #ifndef QOA_NO_STDIO
  
-@@ -394,9 +394,9 @@ unsigned int qoa_encode_frame(const short *sample_data, qoa_desc *qoa, unsigned
- 			#ifdef QOA_RECORD_TOTAL_ERROR
+@@ -168,9 +168,10 @@ void *qoa_read(const char *filename, qoa_desc *qoa);
+ 
+ #ifdef QOA_IMPLEMENTATION
+ #include <stdlib.h>
++#include "core/os/memory.h"
+ 
+ #ifndef QOA_MALLOC
+-	#define QOA_MALLOC(sz) malloc(sz)
++	#define QOA_MALLOC(sz) memalloc(sz)
+ 	#define QOA_FREE(p) free(p)
+ #endif
+ 
+@@ -395,7 +396,7 @@ unsigned int qoa_encode_frame(const short *sample_data, qoa_desc *qoa, unsigned
  				qoa_uint64_t best_error = -1;
  			#endif
--			qoa_uint64_t best_slice;
+ 			qoa_uint64_t best_slice = 0;
 -			qoa_lms_t best_lms;
--			int best_scalefactor;
-+			qoa_uint64_t best_slice = -1;
-+			qoa_lms_t best_lms = {{-1, -1, -1, -1}, {-1, -1, -1, -1}};
-+			int best_scalefactor = -1;
++			qoa_lms_t best_lms = {};
```

Comment:
Are those upstream changes, or part of this fix? If they're upstream changes they shouldn't be in the `.patch` file.

Reply:
Anything not on the `.patch` file is upstream

Reply:
Then there has been no upstream changes? The patch file is exactly what's being added to `qoa.h` in this PR.

Reply:
Previous version didn't initialize `best_slice` and `best_scalefactor` before. Upstream does.

Reply:
Ok I get it now, the diff view on GitHub colors it wrong and makes what actually changed in the patch very confusing.